// Copyright 2023 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package kv_server.application_pa;

import "google/protobuf/struct.proto";

// https://github.com/WICG/turtledove/blob/main/FLEDGE_Key_Value_Server_API.md#query-api-version-2

message KeyGroupOutput {
  repeated string tags = 1;
  map<string, ValueObject> key_values = 2;
}

message ValueObject {
  google.protobuf.Value value = 1;
  int64 global_ttl_sec = 2;
  int64 dedicated_ttl_sec = 3;
}

message KeyGroupOutputs {
  repeated KeyGroupOutput key_group_outputs = 1;
  int32 udf_output_api_version = 2;
}
